src/components/RightPane/RightPaneRoot.tsx

```diff
@@ -35,13 +35,37 @@ class DesktopTabs extends PureComponent<DesktopTabsProps> {
         this.setState({ activeTab: activeTab });
     };
 
+    focusOnBuilding = (buildingInfo: BuildingFocusInfo) => {
+        // If the Map tab isn't already active
+        if (RightPaneStore.getActiveTab() !== 2) {
+            // Switch to Map tab
+            RightPaneStore.handleTabChange(undefined, 2);
```

Comment:
Can you move this to after you add the event listener to make sure there's no race condition?